src/overlays/actors/ovl_Oceff_Wipe/z_oceff_wipe.c

```diff
@@ -27,11 +26,104 @@ const ActorInit Oceff_Wipe_InitVars = {
     (ActorFunc)OceffWipe_Update,
     (ActorFunc)OceffWipe_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Oceff_Wipe/OceffWipe_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Oceff_Wipe/OceffWipe_Destroy.s")
+#include "z_oceff_wipe_gfx.c"
+
+u8 sOceffWipeAlphaIndices[] = {
+    0x01, 0x10, 0x22, 0x01, 0x20, 0x12, 0x01, 0x20, 0x12, 0x01,
+    0x10, 0x22, 0x01, 0x20, 0x12, 0x01, 0x12, 0x21, 0x01, 0x02,
+};
+
+void OceffWipe_Init(Actor* thisx, GlobalContext* globalCtx) {
+    OceffWipe* this = THIS;
+    Actor_SetScale(this, 0.1F);
+    this->counter = 0;
+    this->actor.posRot.pos = ACTIVE_CAM->eye;
+    osSyncPrintf(VT_FGCOL(CYAN) " WIPE arg_data = %d\n" VT_RST, this->actor.params);
+}
+
+void OceffWipe_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    OceffWipe* this = THIS;
+    Player* player = PLAYER;
+
+    func_800876C8(globalCtx);
+    if (gSaveContext.nayrusLoveTimer) {
+        player->unk_692 |= 0x40;
+    }
+}
+
+void OceffWipe_Update(Actor* thisx, GlobalContext* globalCtx) {
+    OceffWipe* this = THIS;
+    this->actor.posRot.pos = ACTIVE_CAM->eye;
+    if (this->counter < 100) {
+        this->counter++;
+    } else {
+        Actor_Kill(this);
+    }
+}
+
+void OceffWipe_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    u32 scroll;
+    OceffWipe* this = THIS;
+    f32 z;
+    GraphicsContext* gfxCtx;
+    u8 alphaTable[3];
+    s32 i;
+    Vec3f eye;
+    Vtx_tn(*vtxPtr)[2];
```

Comment:
I don't think using internal types like `Vtx_tn` is good practice, so I would avoid it unless absolutely necessary.